File: June/24-06-2024/Summed Matrix.cpp
//{ Driver Code Starts

#include <bits/stdc++.h>
using namespace std;

// } Driver Code Ends

class Solution {
  public:
    long long sumMatrix(long long n, long long q) {
        // Your Code Start
           if (q == 1 || q > 2 * n) return 0;
        
        long midCount = n;
        long midElement = n + 1;
        long ans = 0;
        
        if (q >= midElement) {
            ans = midCount - q + midElement;
        } else {
            ans = midCount - midElement + q;
        }
        
        return ans;
      //Your Code End
    }
};

//{ Driver Code Starts.
int main() {
    int t;
    cin >> t;
    while (t--) {
        long long n,q;
        
        cin>>n>>q;

        Solution ob;
        cout << ob.sumMatrix(n,q) << endl;
    }
    return 0;
}
// } Driver Code Ends

File: June/24-06-2024/Summed Matrix.java
//{ Driver Code Starts
// Initial Template for Java

import java.io.*;
import java.util.*;

class GFG {
    public static void main(String args[]) throws IOException {
        BufferedReader read = new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(read.readLine());
        while (t-- > 0) {
            String S[] = read.readLine().split(" ");

            long n = Long.parseLong(S[0]);
            long q = Long.parseLong(S[1]);

            Solution ob = new Solution();
            System.out.println(ob.sumMatrix(n, q));
        }
    }
}
// } Driver Code Ends


// User function Template for Java

class Solution {
    static long sumMatrix(long n, long q) {
        // Your Code Start
            if (q == 1 || q > 2 * n) return 0;
        
        long midCnt = n;
        long midElement = n + 1;
        long ans = 0;
        
        if (q >= midElement) {
            ans = midCnt - q + midElement;
        } else {
            ans = midCnt - midElement + q;
        }
        
        return ans;
      //Your Code End
    }
}
